source/jobs.c

```diff
@@ -536,6 +577,46 @@ static JobsStatus_t matchApi( char * topic,
     return ret;
 }
 
+static bool isThingnameTopicMatch( const char * topic,
+                                   const size_t topicLength,
+                                   const char * topicSuffix,
+                                   const size_t topicSuffixLength,
+                                   const char * thingName,
+                                   const size_t thingNameLength )
+{
+    /* TODO: Inefficient - better implementation shouldn't use snprintf */
+    char expectedTopicBuffer[ TOPIC_BUFFER_SIZE + 1 ] = { 0 };
+    char thingNameBuffer[ MAX_THING_NAME_LENGTH + 1 ] = { 0 };
+    char suffixTerminated[ TOPIC_BUFFER_SIZE - MAX_THING_NAME_LENGTH - 13U ] = { 0 };
+    bool isMatch = true;
+
+    if( ( topic == NULL ) || ( topicLength == 0 ) )
+    {
+        isMatch = false;
+    }
+    else if( ( thingName == NULL ) || ( thingNameLength == 0 ) )
+    {
+        isMatch = false;
+    }
+
+    if( isMatch )
+    {
+        memcpy( thingNameBuffer, thingName, thingNameLength );
+        memcpy( suffixTerminated, topicSuffix, topicSuffixLength );
+        snprintf( expectedTopicBuffer,
```

Comment:
Could we swap this to using the `strAppend` method provided in this file? This would make the code more consistent.

Ref - https://github.com/aws/Jobs-for-AWS-IoT-embedded-sdk/blob/main/source/jobs.c#L189-L228

Reply:
Good point. Swapped them.